src/sage/rings/function_field/function_field.py

```diff
@@ -9,122 +9,126 @@
+    sage: D = P.divisor()                                                               # needs sage.rings.function_field
+    sage: D.basis_function_space()                                                      # needs sage.rings.function_field
     [1]
-    sage: (2*D).basis_function_space()                                                  # optional - sage.rings.function_field
+    sage: (2*D).basis_function_space()                                                  # needs sage.rings.function_field
     [1]
-    sage: (3*D).basis_function_space()                                                  # optional - sage.rings.function_field
+    sage: (3*D).basis_function_space()                                                  # needs sage.rings.function_field
     [1]
-    sage: (4*D).basis_function_space()                                                  # optional - sage.rings.function_field
+    sage: (4*D).basis_function_space()                                                  # needs sage.rings.function_field
     [1, 1/x^4*y + 1/x^4]
 
     sage: K.<x> = FunctionField(QQ); _.<Y> = K[]
-    sage: F.<y> = K.extension(Y^3 - x^2*(x^2 + x + 1)^2)                                # optional - sage.rings.function_field
-    sage: O = F.maximal_order()                                                         # optional - sage.rings.function_field
-    sage: I = O.ideal(y)                                                                # optional - sage.rings.function_field
-    sage: I.divisor()                                                                   # optional - sage.rings.function_field
+    sage: F.<y> = K.extension(Y^3 - x^2*(x^2 + x + 1)^2)                                # needs sage.rings.function_field
+    sage: O = F.maximal_order()
+    sage: I = O.ideal(y)                                                                # needs sage.rings.function_field
+    sage: I.divisor()                                                                   # needs sage.rings.function_field
     2*Place (x, y, (1/(x^3 + x^2 + x))*y^2)
      + 2*Place (x^2 + x + 1, y, (1/(x^3 + x^2 + x))*y^2)
 
     sage: K.<x> = FunctionField(QQ); _.<Y> = K[]
-    sage: L.<y> = K.extension(Y^2 + Y + x + 1/x)                                        # optional - sage.rings.function_field
-    sage: O = L.maximal_order()                                                         # optional - sage.rings.function_field
-    sage: I = O.ideal(y)                                                                # optional - sage.rings.function_field
-    sage: I.divisor()                                                                   # optional - sage.rings.function_field
+    sage: L.<y> = K.extension(Y^2 + Y + x + 1/x)                                        # needs sage.rings.function_field
+    sage: O = L.maximal_order()                                                         # needs sage.rings.function_field
```

Comment:
actually, since we have `# needs sage.rings.function_field`, it could be at the top level of the file, no ?

Reply:
The feature `sage.rings.function_field` indicates that the full function field functionality (in particular, extensions) is available.
`function_field.py` however is shipped with the distribution **sagemath-categories** to make rational function fields available, and that's what a portion of the tests here are testing.